Patch-mainline: Queued in subsystem maintainer repository
Git-repo: https://gitlab.suse.de/coco/tdx/kernel-downstream-suse.git
Git-commit: d8d2fdeae85c905e5d0552ebed6bf78afdf0fd9d
References: jsc#PED-6143
From: Feng Tang <feng.tang@intel.com>
Date: Tue, 8 Oct 2024 19:44:18 +0800
Subject: [PATCH 149/155] KVM: TDX: Fix incompatibility with QEMU definition

about VM time

Otherwise the QEMU startup will fail in early phase.

Signed-off-by: Feng Tang <feng.tang@intel.com>
Signed-off-by: Juergen Gross <jgross@suse.com>
---
 arch/x86/kvm/vmx/main.c         | 1 +
 1 file changed, 1 insertions(+), 0 deletions(-)

diff --git a/arch/x86/kvm/vmx/main.c b/arch/x86/kvm/vmx/main.c
index 4f232b23d91a..249448a5696a 100644
--- a/arch/x86/kvm/vmx/main.c
+++ b/arch/x86/kvm/vmx/main.c
@@ -1215,6 +1215,7 @@ static int __init vt_init(void)
 				  sizeof(struct vcpu_tdx));
 		vcpu_align = max_t(unsigned int, vcpu_align,
 				   __alignof__(struct vcpu_tdx));
+		kvm_caps.supported_vm_types |= BIT(KVM_X86_TDX_VM);
 	}
 	r = kvm_init(vcpu_size, vcpu_align, THIS_MODULE);
 	if (r)
-- 
2.43.0

